packages/platform-android/src/commands/runAndroid/runOnAllDevices.ts

```diff
@@ -27,13 +28,24 @@ function toPascalCase(value: string) {
   return value[0].toUpperCase() + value.slice(1);
 }
 
-function runOnAllDevices(
+async function runOnAllDevices(
   args: Flags,
   cmd: string,
   packageNameWithSuffix: string,
   packageName: string,
   adbPath: string,
 ) {
+  const devices = adb.getDevices(adbPath);
+  if (devices.length === 0) {
+    logger.info('Trying to launch emulator...');
+    const result = await tryLaunchEmulator(adbPath);
+    if (result) {
+      logger.info('Emulator launch succeeded!');
+    } else {
+      logger.warn('Emulator launch failed.');
```

Comment:
Add "Please open an emulator manually, Or connect a device."
Also please add a debug log with the error message